### PR TITLE
Add regex support for pull request links

### PR DESCRIPTION
**Pull request by ephemeral8997**
This pr will add message edition support to the pull request index detecting, and it uses regex.

### CODE REVIEW

**Comment by ephemeral8997** [comment]
The pr now gets the last 10 messages instead, because we do not want to spam the API with useless HTTP calls. Also, there're regexes for pull request link and its index.

**Comment by Dorukyum** [comment]
I think we can put the edit handling on hold for now but the regex search is perfect.

File: cogs/general.py [hunk 1]
```diff
@@ -2,12 +2,16 @@
 from io import StringIO
 from contextlib import suppress
 from urllib import parse
+import re
+import datetime
```
Comment (Dorukyum) [inline cogs/general.py]: Why is datetime imported?

Comment (ephemeral8997) [reply cogs/general.py]: I forgot to remove it when I wanted to use datetimes to determine the message to edit, I'll remove it

**Comment by ephemeral8997** [comment]
The message edition needs a bit improvement, It makes a HTTP request per message edition so It takes time and abuses the API when editing for many times

**Comment by Dorukyum** [comment]
That's why I think we should put it on hold for now.

File: cogs/general.py [hunk 2]
```diff
@@ -2,12 +2,15 @@
...
 from urllib import parse
+import re
 
 import discord
 from discord.ext.commands import Context, command
 
 from utils import Cog
 
+PULL_HASH_REGEX = re.compile(r'##(?P<index>[0-9]+)')
+GITHUB_URL_REGEX = re.compile(r'http(s)?://(www\.)?github\.com/Pycord-Development/pycord/pull/(?P<index>[0-9]+)')
```
Comment (Dorukyum) [inline cogs/general.py]: This is used by gitlink now so we should remove it from here
```suggestion
```

File: cogs/general.py [hunk 3]
```diff
@@ -74,14 +77,46 @@ async def on_message(self, message: discord.Message):
...
+                        link = github_url_format.format(index=after_pull_indexes[before_pull_indexes.index(link)])
+                    except IndexError:
+                        # new pull index added
+                        new_list.append(link)
+                    else:
+                        if link not in new_list:
+                            new_list.append(link)
+            await message.edit(content='\n'.join(new_list))
+            return
+
```
Comment (Dorukyum) [inline cogs/general.py]: I really dont think we should do this, at least not yet
```suggestion
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

File: cogs/general.py (hunk 3 above)
Comment (ephemeral8997) [reply cogs/general.py]: I'll make it as a null docstring to leave it.

Comment (ephemeral8997) [reply cogs/general.py]: or maybe I'll just remove it so we can do it better in another time.

**Pull request merged**